src/fheroes2/editor/editor_ui_helper.cpp

```diff
@@ -153,4 +155,18 @@ namespace Editor
         text.set( std::to_string( resources.gold ), fontType );
         text.draw( thirdColumnOffset + ( maxWidth * 2 + midElementOffsetX - text.width() ) / 2, offsetY[1] + 1, output );
     }
+
+    std::string getDateDescription( const int32_t day )
+    {
+        std::string message = _( "Day: %{day} Week: %{week} Month: %{month}" );
+        int32_t days = day;
+        const int32_t month = ( days - 1 ) / ( 7 * 4 );
+        days -= month * ( 7 * 4 );
+
+        StringReplace( message, "%{day}", ( ( days - 1 ) % 7 ) + 1 );
+        StringReplace( message, "%{week}", ( ( days - 1 ) / 7 ) + 1 );
```

Comment:
We can simplify the calculations a bit here:
```suggestion
        int32_t days = day - 1;
        const int32_t month = days / ( 7 * 4 );
        days -= month * ( 7 * 4 );

        StringReplace( message, "%{day}", ( days % 7 ) + 1 );
        StringReplace( message, "%{week}", ( days / 7 ) + 1 );
```

Reply:
It is done. Thank you for pointing this out!